straxen/plugins/peak_processing.py

```diff
@@ -331,98 +336,393 @@ def find_n_competing(peaks, windows, fraction):
-                 default=('electron_drift_velocity', 'ONLINE', True),
-                 help='Vertical electron drift velocity in cm/ns (1e4 m/ms)'),
-    strax.Option(name='max_drift_length', default=straxen.tpc_z,
-                 help='Total length of the TPC from the bottom of gate to the '
-                      'top of cathode wires [cm]'),
-    strax.Option(name='exclude_drift_time', default=False,
-                 help='Subtract max drift time to avoid peak interference in '
-                      'a single event [ns]'))
 class PeakShadow(strax.OverlapWindowPlugin):
     """
-    This plugin can find and calculate the previous S2 shadow at peak level,
-    with time window backward and previous S2 area as options.
-    It also gives the area and position information of these previous S2s.
+    This plugin can find and calculate the time & position shadow
+    from previous peaks in time.
+    It also gives the area and (x,y) of the previous peaks.
+    References:
+        * v0.1.3 reference: xenon:xenonnt:ac:prediction:shadow_ambience
     """
 
-    __version__ = '0.1.0'
+    __version__ = '0.1.3'
     depends_on = ('peak_basics', 'peak_positions')
     provides = 'peak_shadow'
     save_when = strax.SaveWhen.EXPLICIT
 
-    def setup(self):
-        self.time_window_backward = self.config['time_window_backward']
-        if self.config['exclude_drift_time']:
-            electron_drift_velocity = straxen.get_correction_from_cmt(
-                self.run_id,
-                self.config['electron_drift_velocity'])
-            drift_time_max = int(self.config['max_drift_length'] / electron_drift_velocity)
-            self.n_drift_time = drift_time_max
-        else:
-            self.n_drift_time = 0
-        self.s2_threshold = self.config['pre_s2_area_threshold']
-        self.exponent = self.config['deltatime_exponent']
+    shadow_time_window_backward = straxen.URLConfig(
+        default=int(1e9),
+        type=int,
+        track=True,
+        help='Search for peaks casting time & position shadow in this time window [ns]'
+    )
+
+    shadow_threshold = straxen.URLConfig(
+        default={'s1_time_shadow' : 1e3, 
+                 's2_time_shadow' : 1e4, 
+                 's2_position_shadow' : 1e4},
+        type=dict,
+        track=True,
+        help='Only take S1/S2s larger than this into account when calculating Shadow [PE]'
+    )
+
+    shadow_deltatime_exponent = straxen.URLConfig(
+        default=-1.0,
+        type=float,
+        track=True,
+        help='The exponent of delta t when calculating shadow'
+    )
+
+    shadow_sigma_and_baseline = straxen.URLConfig(
+        default=[15.220, 0.036],
+        type=list,
+        track=True,
+        help='Fitted position correlation sigma[cm*PE^0.5] and baseline[cm] using in position shadow'
+    )
 
     def get_window_size(self):
-        return 3 * self.config['time_window_backward']
+        return 10 * self.config['shadow_time_window_backward']
```

Comment:
sorry @dachengx, did you change time window from 3sec to 10sec? Do you have any notes for this change (sorry I maybe missed your updates)?

Reply:
Hey @skazama . This function is related to [this](https://strax.readthedocs.io/en/latest/developer/overlaps.html) doc and it is fully technical. The time window is still 1 sec. 3 & 10 will give the same results here.